### ISSUE TITLE
[no-unused-var] rule ignores underscore (intentional unused var)

### ISSUE THREAD

**Issue by darklight9811** [typescript-eslint/typescript-eslint#3324]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": ["<any>"]
  }
}
```

```TS
function test (_, _ignore) {
// any code
}
```

**Expected Result**
Since this is a standard when programming, variables initialized with an underscore are to be disconsidered since they are there to access later arguments or just to quickly type it. They should not be considered unused since this is already intentional behavior, neither they should complain about any implicitly, since they are not going to be used.

**Actual Result**
![image](https://user-images.githubusercontent.com/20667774/116427613-8e015480-a81a-11eb-9018-69b6e04407c7.png)

**Additional Info**

I know this is configurable with patterns, but this should already come out of the box since in any language and any framework, underscore has this behavior.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.22.0` |
| `@typescript-eslint/parser`        | `4.3.0` |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.15.0` |
| `node`                             | `14.3.0` |

**Comment by bradzacher** [comment]
> I know this is configurable with patterns, but this should already come out of the box since in any language and any framework, underscore has this behavior.

We mirror [the base rule's](https://eslint.org/docs/rules/no-unused-vars) default configuration. By default it provides no ignore default, we don't either.
